### PR TITLE
AN-92 Link refinements

### PR DESCRIPTION
**Pull request by kevinfodness**
* Revise link handling to address several edge cases of empty or malformed URLs, properly handle root-relative URLs, and properly handle anchor links.
* Add test cases to check each of these.
* Fix tests as a result of a previous PR that sets defaults for `isPreview` etc.

### CODE REVIEW

File: includes/apple-exporter/class-parser.php [hunk 1]
```diff
@@ -146,53 +146,62 @@ private function _parse_markdown( $html ) {
...
 
-		// Iterate over the matches and see what we need to do
-		foreach ( $a_tags[1] as $i => $tag_type ) {
-			// First, if this is an anchor, those aren't supported so just remove it.
-			if ( 'name' === $tag_type ) {
-				$html = str_replace( $a_tags[0][ $i ], $a_tags[3][ $i ], $html );
+		// Iterate over the matches and see what we need to do.
+		foreach ( $a_tags[0] as $i => $a_tag ) {
+			// If the <a> tag doesn't have content, dump it.
+			$content = trim( $a_tags[1][ $i ] );
```
Comment (mslinnea) [inline includes/apple-exporter/class-parser.php]: I may be reading this incorrectly, but I'm wondering if this should be `$a_tags[0][ $i ]` instead?

File: includes/apple-exporter/class-parser.php [hunk 2]
```diff
@@ -146,53 +146,62 @@ private function _parse_markdown( $html ) {
...
-				if ( $href !== $a_tags[2][ $i ] ) {
-					$updated_tag = str_replace( $a_tags[2][ $i ], $href, $a_tags[0][ $i ] );
-					$html = str_replace( $a_tags[0][ $i ], $updated_tag, $html );
-				}
+				$html = str_replace( 'href="' . $href, 'href="' . $permalink . $href, $html );
+				continue;
+			}
+
+			// Handle root relative URLs.
+			if ( 0 === strpos( $href, '/' ) && false === strpos( $href, '//' ) ) {
```
Comment (mslinnea) [inline includes/apple-exporter/class-parser.php]: good thinking on checking for `//`!

File: includes/apple-exporter/class-parser.php (hunk 1 above)
Comment (kevinfodness) [reply includes/apple-exporter/class-parser.php]: `[0]` is going to contain the full match, but `[1]` will contain the content of the first capture group, which in this case is what is inside of the `a` tag. Personally, I think the PHP authors got the organization of the array backwards and that it's confusing, but so it goes :)
